q2_demux/_summarize/_visualizer.py

```diff
@@ -162,6 +162,8 @@ def summarize(output_dir: str, data: _PlotQualView, n: int = 10000) -> None:
         quality_scores, min_seq_len = _subsample_single(sample_map)
 
     forward_scores = pd.DataFrame(quality_scores['forward'])
+    if forward_scores.empty:
```

Comment:
I think it might make more sense to move this check up higher, say ~L145 ---

```python
if sequence_count == 0:
    raise ValueError('No sequences present.')
```

Or, we could skip the `ValueError` altogether and update the rendering of the viz to skip rendering the quality score plots tab. I think I like that idea more.

Reply:
Sorry for my delayed reply, this one kinda slipped through the cracks for me, and I just noticed this latest comment. I don't think the issue referenced in @ChrisKeefe's issue #103 is about having no quality scores, his example artifact contains no reads at all period. Unless I'm misunderstanding something (or put the error message in a place that would imply only the quality scores were missing, which is what it looks like I did) I don't think there would be anything to render at all given his artifact would there? All the same, this may have unearthed a new improvement. If you want I can both improve the error message if there are no reads at all, and simply skip rendering the quality score plots tab if we are only missing quality scores. Would we want to user to be alerted that there were quality scores missing? Or is this `summarize` method also meant to be used on `fasta` files where no quality scores are expected? I'm not really sure how that would work without excepting given the current state of the code though.

Reply:
Additionally, the proposed check doesn't catch the case where there are reads in the forward file and no reads in the reverse file. Is this case actually problematic, or did I incorrectly assume it is?

Reply:
drive-by cc: @thermokarst

Reply:
@Oddant1 - 

> I don't think the issue referenced in @ChrisKeefe's issue #103 is about having no quality scores, his example artifact contains no reads at all period

Yep, the original issue doesn't reference quality scores, although _your proposed fix does_. What the proposed fix doesn't actually do is check (directly) for an empty table.

> I don't think there would be anything to render at all given his artifact would there?

We can render the first tab of the visualization:

![image](https://user-images.githubusercontent.com/274668/63452380-02a3af00-c3fb-11e9-9e49-4f7277ddf161.png)

Seeing a bunch of "zeros" in there would actually be pretty informative for users.

> If you want I can both improve the error message if there are no reads at all,

I don't think we should raise an Exception --- you can move the warning into the viz itself though (as a banner at the top of the first tab).

> Or is this summarize method also meant to be used on fasta files where no quality scores are expected?

It is not meant for FASTA files (see my note above about quality scores as part of your solution).